src/BBitsRaffle.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {ReentrancyGuard} from "@openzeppelin/utils/ReentrancyGuard.sol";
+import {Ownable} from "@openzeppelin/access/Ownable.sol";
+import {Pausable} from "@openzeppelin/utils/Pausable.sol";
+import {IERC721} from "@openzeppelin/token/ERC721/IERC721.sol";
+import {IBBitsCheckIn} from "./interfaces/IBBitsCheckIn.sol";
+import {IBBitsRaffle} from "./interfaces/IBBitsRaffle.sol";
+
+/// @title  Based Bits Raffle
+/// @notice This contract ...
+contract BBitsRaffle is IBBitsRaffle, Ownable, ReentrancyGuard, Pausable {
+    /// @notice Based Bits NFT.
+    IERC721 public immutable collection;
+
+    /// @notice Based Bits checkin contract.
+    IBBitsCheckIn public immutable checkIn;
+
+    /// @notice The current raffle status of this contract.
+    ///         PendingRaffle:     Raffle not started (just deployed/just settled).
+    ///         InRaffle:          Accepting entries.
+    ///         PendingSettlement: No longer accepting entries but not settled.
+    RaffleStatus public status;
+
+    /// @notice The total number of raffles held by this contract.
+    uint256 public raffleCount;
```

Comment:
Safer to leave as a uint256 in case raffle duration is reduced to a short time period - unlikely to surpass (2^16 - 1 = 65535) days, but feasible if duration is set to hours/minutes.